clab/clab_test.go

```diff
@@ -46,14 +55,23 @@ func Test_createStaticDynamicDependency(t *testing.T) {
 	// retrieve a map of nodes
 	nodeMap := getNodeMap(mockCtrl)
 
-	dm.EXPECT().AddDependency("node4", "node1")
-	dm.EXPECT().AddDependency("node4", "node2")
-	dm.EXPECT().AddDependency("node4", "node3")
-	dm.EXPECT().AddDependency("node5", "node1")
-	dm.EXPECT().AddDependency("node5", "node2")
-	dm.EXPECT().AddDependency("node5", "node3")
+	clab := &CLab{
+		Nodes: nodeMap,
+	}
+
+	err := WithDependencyManager(dm)(clab)
```

Comment:
why not `clab.WithDependencyManager(dm)`?

Reply:
`WithDependencyManager ` yields a `ClabOption`. 
`ClabOption` is defined in the following way:
```
type ClabOption func(c *CLab) error
```
So it will apply itself on the clab instance.
Hence we're also checking the `WithDependencyManager ` option. A `clab.WithDependencyManager(dm)` does not exist.
We could only set the option `manually` via:
```
	clab := &CLab{
		Nodes: nodeMap,
		dependencyManager: dm,
	}
```
But this way it makes perfect sense to me.